OPAL/br/src/main/scala/org/opalj/br/fpcf/analyses/FieldAccessInformationAnalysis.scala

```diff
@@ -0,0 +1,152 @@
+ *
+ * @note Fields which are not accessed at all are not further considered.
+ *
+ * @author Maximilian Rüsch
+ */
+class FieldAccessInformationAnalysis(val project: SomeProject) extends FPCFAnalysis {
+
+    private val declaredMethods = project.get(DeclaredMethodsKey)
+
+    def analyzeMethod(method: Method): PropertyComputationResult = {
+        implicit val logContext: LogContext = project.logContext
+        import project.resolveFieldReference
+
+        val readAccesses = mutable.AnyRefMap.empty[Field, IntTrieSetBuilder]
+        val writeAccesses = mutable.AnyRefMap.empty[Field, IntTrieSetBuilder]
+
+        // we don't want to report unresolvable field references multiple times
+        val reportedFieldAccesses = ConcurrentHashMap.newKeySet[Instruction]()
+
+        method.body.get iterate { (pc, instruction) =>
+            instruction.opcode match {
+                case GETFIELD.opcode | GETSTATIC.opcode =>
+                    val fieldReadAccess = instruction.asInstanceOf[FieldReadAccess]
+                    resolveFieldReference(fieldReadAccess) match {
+                        case Some(field) =>
+                            readAccesses.getOrElseUpdate(field, new IntTrieSetBuilder()) += pc
+                        case None =>
+                            if (reportedFieldAccesses.add(instruction)) {
+                                val message = s"cannot resolve field read access: $instruction"
+                                OPALLogger.warn("project configuration", message)
+                            }
+                        // IMPROVE: In the old implementation, unresolvable field accesses were tracked. This can
+                        // also be done here but is not needed as of now.
+                    }
+
+                case PUTFIELD.opcode | PUTSTATIC.opcode =>
+                    val fieldWriteAccess = instruction.asInstanceOf[FieldWriteAccess]
+                    resolveFieldReference(fieldWriteAccess) match {
+                        case Some(field) =>
+                            writeAccesses.getOrElseUpdate(field, new IntTrieSetBuilder()) += pc
+                        case None =>
+                            if (reportedFieldAccesses.add(instruction)) {
+                                val message = s"cannot resolve field write access: $instruction"
+                                OPALLogger.warn("project configuration", message)
+                            }
+                        // IMPROVE: In the old implementation, unresolvable field accesses were tracked. This can
+                        // also be done here but is not needed as of now.
+                    }
+
+                case _ => /*nothing to do*/
+            }
+        }
+
+        val definedMethod = declaredMethods(method)
+        val fieldAccesses = mutable.AnyRefMap.empty[Field, (List[(DefinedMethod, PCs)], List[(DefinedMethod, PCs)])]
```

Comment:
What is the reason for having this intermediate data structure? Can't we just use the readAcceses/writeAcceses when creating the Result? If that is not yet possible, can we change the collection logic above so that it becomes possible?